src/sentry/integrations/slack/utils/channel.py

```diff
@@ -173,3 +178,115 @@ def get_channel_id_with_timeout(
             return id_data
 
     return prefix, None, False
+
+
+def get_channel_id_with_timeout_new(
+    integration: Integration | RpcIntegration,
+    name: Optional[str],
+    timeout: int,
+) -> Tuple[str, Optional[str], bool]:
+    """
+    Fetches the internal slack id of a channel.
+    :param integration: The slack integration
+    :param name: The name of the channel
+    :param timeout: Our self-imposed time limit.
+    :return: a tuple of three values
+        1. prefix: string (`"#"` or `"@"`)
+        2. channel_id: string or `None`
+        3. timed_out: boolean (whether we hit our self-imposed time limit)
+    """
+
+    payload = {
+        "exclude_archived": False,
+        "exclude_members": True,
+        "types": "public_channel,private_channel",
+    }
+
+    time_to_quit = time.time() + timeout
+
+    client = SlackClient(integration_id=integration.id)
+    id_data: Optional[Tuple[str, Optional[str], bool]] = None
+    found_duplicate = False
+    prefix = ""
+    channel_id = None
+    try:  # Check for channel
+        channel_id = check_for_channel(client, name)
+        prefix = "#"
+    except ApiError as e:
+        if str(e) != "channel_not_found":
+            # return prefix, None, False
+            raise e
+        # Check if user
+        while True:
+            cursor = ""
+            # Slack limits the response of `<list_type>.list` to 1000 channels
+            try:
+                items = client.get("/users.list", params=dict(payload, cursor=cursor, limit=1000))
+            except ApiRateLimitedError as e:
+                logger.info("rule.slack.user_list_rate_limited", extra={"error": str(e)})
+                raise e
+            except ApiError as e:
+                logger.info("rule.slack.user_list_rate_limited", extra={"error": str(e)})
```

Comment:
Hmm I know you copied this from the old function but the logger messages shouldn't be the same here. If it's an `ApiError` it's not rate limited, right? You can change this to like `"rule.slack.user_list_api_error"`.